### PR TITLE
Adding new Air Circular 20-172 traffic icon mode, to make icons more …

### PR DESCRIPTION
**Pull request by shanelenagh**
…visible and consistent with the ADSB/TCAS standards.  This PR includes the following items:

1. New AC 20-172 icons, nominally cyan triangles with a solid black outline.  When traffic gets closer the color changes, and overlays appear with an orange circle "advisory" (very close traffic) status and  a red square for "resolution" (closing with a time of closest point of approach within a critical range) status.
2. The relative flight level and ascending/descending status are indicated by a box to the right of the traffic icon showing the relative flight level in hundreds of feet, similar to classic Avare, with and up (🠉) or down (🠋) arrow if the traffic is ascending or descending.
3. Fixed the Windows crash of the audible alerts by implementing a workaround fix (until they fix the mainline package) for the MSIX package, per this issue report: https://github.com/YehudaKremer/msix/issues/272#issuecomment-2181634105
4. Made the audible alerts muted icon the standard Material "volume off" icon we previously discussed, in a semi-transparent white paint (clutters the UI less and has clearer intent than the black X, IMHO).  Let me know what you think.

I am sending you a video walkthrough of the code and features.  Sorry it took so long to do this.

### CODE REVIEW

File: .github/workflows/windows.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
       - uses: subosito/flutter-action@v2
         with:
           channel: 'stable'
-          flutter-version: '3.22.2'
+          flutter-version: '3.22.3'
```
Comment (apps4av) [inline .github/workflows/windows.yaml:16]: Is this required? Because action script uses version .2

File: lib/gdl90/traffic_alerts.dart [hunk 2]
```diff
@@ -15,6 +15,9 @@ enum DistanceCalloutOption { none, rounded, decimal }
 
 enum NumberFormatOption { colloquial, individualDigit }
 
+const _mToFt = 3.28084;
+const _mpsToKt = 1.94384;
```
Comment (apps4av) [inline lib/gdl90/traffic_alerts.dart:19]: Use units_convesion dart file and see windsaloft.dart for example

File: .github/workflows/windows.yaml (hunk 1 above)
Comment (shanelenagh) [reply .github/workflows/windows.yaml]: I committed a downgrade (saw a message in part of my Windows debugging that steered me to that version, but I'm giving the lower one a shot right now)

File: lib/gdl90/traffic_alerts.dart (hunk 2 above)
Comment (shanelenagh) [reply lib/gdl90/traffic_alerts.dart]: Done

**Comment by n251ea** [comment]
Cell service at OSH is hit it mid as the system is overloaded, I won't be able to view the video until either I get lucky or I leave, but I am really looking forward to seeing the fruits of your labor! Zubair, why statute miles?  Or is that just a setup to allow conversion back and forth between statute and nautical?  Realistically at 5 miles of less the difference is so small it doesn't make a difference. JeffSent from my cell phone 
-------- Original message --------From: shanelenagh ***@***.***> Date: 7/22/24  8:44 PM  (GMT-06:00) To: apps4av/avarex ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [apps4av/avarex] Adding new Air Circular 20-172 traffic icon mode, to make icons more … (PR #7) 
@shanelenagh commented on this pull request.



In lib/gdl90/traffic_alerts.dart:
> @@ -15,6 +15,9 @@ enum DistanceCalloutOption { none, rounded, decimal }
 
 enum NumberFormatOption { colloquial, individualDigit }
 
+const _mToFt = 3.28084;
+const _mpsToKt = 1.94384;

Done

**Comment by shanelenagh** [comment]
@apps4av , hold off on this a minute, as I am seeing that the up/down arrows (which are Unicode characters) for the ascending/descending notation aren't showing up on my Android phone.  So I might need to rethink how that is done.  Guessing there isn't a font to display them by default--might need to resort to an icon or another approach for that.

**Comment by shanelenagh** [comment]
> @apps4av , hold off on this a minute, as I am seeing that the up/down arrows (which are Unicode characters) for the ascending/descending notation aren't showing up on my Android phone. So I might need to rethink how that is done. Guessing there isn't a font to display them by default--might need to resort to an icon or another approach for that.

I have addressed this now--I used a different character that does show up on my phone now.

**Review by apps4av — APPROVED** [review]

**Comment by shanelenagh** [comment]
Just curious, are you (still) at Oshkosh? I came again this year, for the
first time in 3 years (only my second time overall).

**Comment by shanelenagh** [comment]
Sorry, this was meant to go to n251ea (Jeff). Looks like I missed him,
though:

**Pull request merged**